### PR TITLE
hackeries

### PR DESCRIPTION
**Pull request by mark-radai**

### CODE REVIEW

File: tensor2tensor/utils/t2t_model.py [hunk 1]
```diff
@@ -320,7 +321,9 @@ def call(self, inputs, **kwargs):
       summarize_features(features, num_shards=self._num_datashards)
       sharded_features = self._shard_features(features)
       sharded_logits, losses = self.model_fn_sharded(sharded_features)
-      if isinstance(sharded_logits, dict):
+      if (sharded_logits[0].__class__.__name__ == 'HideDict'):
+        sharded_logits = sharded_logits[0].d
```
Comment (epurdy) [inline tensor2tensor/utils/t2t_model.py:326]: Let's assert here that the length of sharded_logits is 1, just to be safe.

File: tensor2tensor/utils/t2t_model.py [hunk 2]
```diff
@@ -320,7 +321,9 @@ def call(self, inputs, **kwargs):
...
+      if isinstance(sharded_logits, dict) or (sharded_logits.__class__.__name__ == 'HideDict'):
```
Comment (epurdy) [inline tensor2tensor/utils/t2t_model.py]: I think this line can go back to the original? We've already mutated sharded_logits into a normal dict.

File: tensor2tensor/utils/t2t_model.py (hunk 1 above)
Comment (mark-radai) [reply tensor2tensor/utils/t2t_model.py:326]: k

File: tensor2tensor/utils/t2t_model.py (hunk 2 above)
Comment (mark-radai) [reply tensor2tensor/utils/t2t_model.py]: valid

**Review by epurdy — APPROVED** [review]
LGTM

**Pull request merged**